coding-guides/tracking-used-invites.md

```diff
@@ -8,48 +8,73 @@ description: >-
 
-To get around this, we need to do a couple separate steps. The first one is to go into your bots application page and enable the `GUILD_MEMBERS` privileged intent and add it to your bots client, then the second thing is to fetch all of the invites for each guild and store it in a temporary object. The third is to add, and remove invites from the object via the `inviteCreate` and `inviteDelete` events and finally is to fetch a guild's invite whenever someone joins, and find which one has a new use on it.
+To get around this, we need to do two separate steps. The first one is to fetch all of the invites for each guild and store it in a temporary Map. The second is to fetch a guild's invite whenever someone joins, and find which one has a new use on it. Thankfully, that's actually pretty simple!
 
-Thankfully, that's actually pretty simple!
+{% hint style="info" %}
+If you don't know what a Map is, read up [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)
+{% endhint %}
 
-While the below is a fair approximation of invite tracking, it's still not perfect. As it doesn't track single use invites \(when right-clicking someone, and doing Invite To =&gt; Server\). Those exist only for a few moments and cannot be tracked at all.
+While the below is a fair approximation of invite tracking, it's still not perfect. There are 2 things it doesn't track:
+
+* Temporary one-use invites \(when right-clicking someone, and doing Invite To =&gt; Server\). Those exist only for a few moments and cannot be tracked at all.
+* Invites created after the bot loaded. That would require fetching on a loop which is dangerous for API spam.
 
 ## Caching Invites
 
-The first part is to fetch all the invites and keep them cached in an object. This is done in the `ready` event. First, however, we must ensure that the cache is initialized _outside_ of the ready event, be it further up in your main file.
+The first part is to fetch all the invites and keep them cached in an Map. This is done in the `ready` event. First, however, we must ensure that the cache is initialized _outside_ of the ready event.
 
 ```javascript
 // Initialize the invite cache
-const invites = {};
+const invites = new Map();
 
 // A pretty useful method to create a delay without blocking the whole script.
-const wait = require("timers/promisify").setTimeout;
+const wait = require("timers/promises").setTimeout;
 
 client.on("ready", async () => {
   // "ready" isn't really ready. We need to wait a spell.
   await wait(1000);
-  // Load all invites for all guilds and save them to the cache.
-  client.guilds.cache.forEach(g => {
-    g.invites.fetch().then(guildInvites => {
-      invites[g.id] = guildInvites;
-    });
+
+  // Loop over all the guilds
+  client.guilds.cache.map(async (guild) => {
+    // Fetch all Guild Invites
+    const firstInvites = guild.invites.fetch();
```

Comment:
If not using `Promise.all` this could be changed to `forEach`. Also missing `await` on invite fetching.
```suggestion
  client.guilds.cache.forEach(async (guild) => {
    // Fetch all Guild Invites
    const firstInvites = await guild.invites.fetch();
```

Reply:
Fixed!